pkg/node/types/node.go

```diff
@@ -351,6 +351,21 @@ func (n *Node) GetK8sNodeIP() net.IP {
 	return externalIP
 }
 
+// GetNodeInternalIP returns the Internal IP of node or nil.
+func (n *Node) GetNodeInternalIP(ipv6 bool) net.IP {
```

Comment:
The boolean flag in`GetNodeInternalIP(true)` and `GetNodeInternalIP(false)` is not so straightforward, I suggest to add two separate helpers (one for IPv4 and another one for IPv6) and get rid of it, so to improve the readability.
Also, that may simplify the `if` condition inside the helper itself.

Reply:
sure will update it. I just followed existing api semantics `GetNodeIP` and `GetCiliumInternalIP` https://github.com/cilium/cilium/blob/6ba173d9ba392e05ff58930b482d697e00c70805/pkg/node/types/node.go#L356